lib/gitWorkflow.js

```diff
@@ -103,7 +103,17 @@ export class GitWorkflow {
       ctx.errors.add(GetBackupStashError)
       throw new Error('lint-staged automatic backup is missing!')
     }
-    return `refs/stash@{${index}}`
+
+    /**
+     * https://github.com/okonet/lint-staged/issues/1121
+     * Detect MSYS shell and escape braces
+     * to prevent interpolation
+     */
+    if (process.env.MSYSTEM === 'MSYS') {
+      return `refs/stash@\\{${index}\\}`
+    } else {
+      return `refs/stash@{${index}}`
+    }
```

Comment:
We can take advantage of "early return" here and skip the `else`:

```suggestion
    }

    return `refs/stash@{${index}}`
```